### PR TITLE
Add subscriber endpoints

### PR DESCRIPTION
**Pull request by jdtoombs**
Ended up fixing the bugs I was facing - putting this up first again.

Add separate endpoints and hooks for the subscriber application.

### CODE REVIEW

File: api/net/Areas/Subscriber/Controllers/ContentController.cs [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+    /// Publish message to kafka to index content in elasticsearch.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    [HttpPost]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(ContentModel), (int)HttpStatusCode.Created)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Content" })]
+    public async Task<IActionResult> AddAsync(ContentModel model)
```
Comment (Fosol) [inline api/net/Areas/Subscriber/Controllers/ContentController.cs:158]: Subscriber's cannot add content

File: api/net/Areas/Subscriber/Controllers/ContentController.cs [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    /// Publish message to kafka to remove content from elasticsearch.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    [HttpDelete("{id}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(ContentModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Content" })]
+    public async Task<IActionResult> DeleteAsync(ContentModel model)
```
Comment (Fosol) [inline api/net/Areas/Subscriber/Controllers/ContentController.cs:330]: Subscribers cannot delete content

File: libs/net/dal/Services/ContentService.cs [hunk 3]
```diff
@@ -220,6 +220,9 @@ public async Task<IPaged<Content>> FindWithElasticsearchAsync(ContentFilter filt
         if (!string.IsNullOrWhiteSpace(filter.Headline))
             filterQueries.Add(s => s.Wildcard(m => m.Field(p => p.Headline).Value($"*{filter.Headline.ToLower()}*")));
 
+        // if (!string.IsNullOrWhiteSpace(filter.Keyword))
```
Comment (Fosol) [inline libs/net/dal/Services/ContentService.cs:223]: You haven't figured out how to make this request yet correct?  If so add a `// TODO: description`

File: libs/net/models/Areas/Subscriber/Models/User/UserModel.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
+using TNO.Entities;
+
+namespace TNO.API.Areas.Subscriber.Models;
+
+/// <summary>
+/// UserModel class, provides a model that represents an user.
+/// </summary>
+public class UserModel
```
Comment (Fosol) [inline libs/net/models/Areas/Subscriber/Models/User/UserModel.cs:8]: Subscribers should not need this

File: libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      return api.get<IPaged<IContentModel>, AxiosResponse<IPaged<IContentModel>>, any>(
+        `/subscriber/contents?${toQueryString(params)}`,
+      );
+    },
+    getContent: (id: number) => {
+      return api.get<IContentModel | undefined, AxiosResponse<IContentModel | undefined>, any>(
+        `/subscriber/contents/${id}`,
+      );
+    },
+    updateContent: (content: IContentModel) => {
```
Comment (Fosol) [inline libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts:44]: Subscribers cannot update content

File: libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      return api.put<IContentModel, AxiosResponse<IContentModel>, any>(
+        `/subscriber/contents/${content.id}`,
+        content,
+      );
+    },
+    updateContentList: (action: IContentListModel) => {
```
Comment (Fosol) [inline libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts:50]: Subscribers cannot update content

**Review by Fosol — CHANGES_REQUESTED** [review]

File: libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts (hunk 5 above)
Comment (jdtoombs) [reply libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts:44]: I vaguely remember seeing on Bobbi's mockup a way to edit a headline.. I may be mistaken

Comment (Fosol) [reply libs/npm/core/src/hooks/api-subscriber/useApiSubscriberContents.ts:44]: Subscribers cannot edit any content.

When creating reports they can edit the report data only.  We don't currently have a way to do this.

File: libs/net/models/Areas/Subscriber/Models/User/UserModel.cs (hunk 4 above)
Comment (jdtoombs) [reply libs/net/models/Areas/Subscriber/Models/User/UserModel.cs:8]: On the content model it is used for getting the owner, is this something we want to keep?

Comment (Fosol) [reply libs/net/models/Areas/Subscriber/Models/User/UserModel.cs:8]: Subscriber's don't care which editor currently owns the content.  It's a feature we built in to enable assigning and identifying who worked on content.

File: api/net/Areas/Subscriber/Models/Storage/FolderModel.cs [hunk 7]
```diff
@@ -0,0 +1,97 @@
+using TNO.Core.Extensions;
+
+namespace TNO.API.Areas.Subscriber.Models.Storage;
+
+/// <summary>
+/// FolderModel class, provides a model that represents an file or directory.
+/// </summary>
+public class FolderModel
```
[the remark this answers is not in the document]
Comment (jdtoombs) [reply api/net/Areas/Subscriber/Models/Storage/FolderModel.cs:8]: Will remove for now, easy to add back if needed

File: api/net/Areas/Subscriber/Controllers/ContentController.cs [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+    /// Publish message to kafka to index content in elasticsearch.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    [HttpPut("{id}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(ContentModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "Content" })]
+    public async Task<IActionResult> UpdateAsync(ContentModel model)
```
[the remark this answers is not in the document]
Comment (Fosol) [reply api/net/Areas/Subscriber/Controllers/ContentController.cs:157]: Still needs to be removed.

**Review by Fosol — CHANGES_REQUESTED** [review]

**Review by Fosol — APPROVED** [review]

**Pull request merged**